genai-perf/genai_perf/inputs/converters/rankings_converter.py

```diff
@@ -52,7 +52,7 @@ def convert(self, generic_dataset: GenericDataset, config: InputsConfig) -> Dict
         request_body: Dict[str, Any] = {"data": []}
 
         for query_index, query_row in enumerate(queries_data.rows):
-            if query_index > rows_of_passage_data:
+            if query_index >= rows_of_passage_data:
```

Comment:
Was this a bug fix?

Reply:
Yes. I forgot Python is 0-indexes. The index needs to be greater or equal to the number of rows to break. As an example, if there are two queries, the second query row would have an index of 1... if there was one passage, the rows_of_passage_data would be 1... so it would not break (1 == 1) and would try to find a second passage then be out of bounds.

This was caught by the [new rankings test](https://github.com/triton-inference-server/perf_analyzer/pull/142/files#diff-047e86ccbac2c3fc8f4e3d037bc584ed6d49f0ea3bde4c21123ffb1d35213684R276) that checks mismatched numbers of queries and passage.